sql-plugin/src/main/scala/com/nvidia/spark/RebaseHelper.scala

```diff
@@ -50,10 +51,13 @@ object RebaseHelper extends Arm {
     }
   }
 
-  def isDateTimeRebaseNeededWrite(column: ColumnVector): Boolean =
-    isDateTimeRebaseNeeded(column,
-      RebaseDateTime.lastSwitchGregorianDay,
-      RebaseDateTime.lastSwitchGregorianTs)
+  def isDateTimeRebaseNeededWrite(column: ColumnVector): Boolean = {
```

Comment:
Why not just do the conversion after the rebase check?  That is what spark does and it feels like it is going to be more maintainable than trying to have this potentially support multiple different column types.

Reply:
I believe we are doing the conversion after. I think your comment about the EXCEPTION path confused me. Let me revert and lets take a look at it again. I am hoping the github will show it properly after that